crates/containerd-shim-wasm/src/container/context.rs

```diff
@@ -44,6 +46,35 @@ pub enum Source<'a> {
     Oci(&'a [WasmLayer]),
 }
 
+impl<'a> Source<'a> {
+    /// Returns the raw bytes of the WASM binary.
+    ///
+    /// If the source is a file, the file is read from the file system.
+    /// If the source is an OCI layer, the layer is read from the OCI spec.
+    ///
+    /// At the moment, only a single module is supported when using images with OCI layers.
+    pub fn into_wasm_binary(self) -> Result<Vec<u8>, anyhow::Error> {
```

Comment:
nice, I think it is still valauble to have some sense where the layers come from with the source? But I like this since, we could use this as the api to that hides composing together multiple components.